lib/tasks.js

```diff
@@ -95,36 +95,77 @@ module.exports = function(req,res) {
         return arySQL;
     };
 
-    this.applyTransaction = (obj) => {
+    this.applyTransaction = (obj, rejectWithError = false) => {
```

Comment:
optional: @itsDiwaker what do you think about instead of adding a new param, we reuse the `obj`? since it's already an `Object`, we can pass the same flag as a new property. I'm thinking it feels and reads better.

```js
// current implementation
await this.restful.applyTransaction(transactionBody, true);
// or 
const rejectWithError = true;
await this.restful.applyTransaction(transactionBody, rejectWithError);

// my personal preference
await this.restful.applyTransaction({
	// other props for transactionPayload,
	rejectWithError: true
});
```

Reply:
@onel0p3z did give it a thought. Had 2 reasons why I didn't go with that approach -

* Having an explicit argument is self-documenting.
* Piggybacking `rejectWithError ` risks it being overridden unexpectedly while writing a complex logic.

Reply:
@itsDiwaker the possibility of this flag being overwritten was one of my first thoughts when I suggested the change but I checked the only time we're using the `obj` variable is when calling `extractSQL` in line 104 so the surface for this to happen is restricted to sections where we're building this object before `applyTransaction` is invoked. 

I disagree this is self-documenting as you'd need to know the function definition to know what the 2nd param is related to unless you use the following:
```js
const rejectWithError = true;
await this.restful.applyTransaction(transactionBody, rejectWithError);
```

Reply:
> I disagree this is self-documenting as you'd need to know the function definition to know what the 2nd param is related to unless you use the following:

@onel0p3z It's self-documenting if we look at the method signature before planning to use it for the first time. It would automatically make you think, what would happen if one is to flip the sign without taking the pain to read through the logic to reverse engineer the fact that they can actually send `obj.rejectWithError` to change the behaviour in my opinion.

`this.applyTransaction = (obj, rejectWithError = false) => {` 

vs

`this.applyTransaction = (obj) => {`

Reply:
> this.applyTransaction = (obj) => { //not apparent at all

```js
/**
* @param {Object} obj - options
* // other params
* @param {Boolean} [obj.rejectWithError] - if enabled, it'll reject with error. Defaults to false. 
*/
this.applyTransaction = (obj) => {
```

for this approach, we'd add a JSDoc block to document the available params as we usually do in other parts of the codebase but I understand this repository does not follow our coding guidelines. I'll resolve the conversation so we can proceed with merging these changes. thank you @itsDiwaker!